doc/refactoring_design.md

```diff
@@ -0,0 +1,481 @@
+* **Tribe**: it controls the registry of membership for each service agreement template; defines TCR based registries for permission, which has whitelisted members who can provide or consume the service.
+* **Curation**: users can curate a registry of high-quality service agreement templates through staking on corresponding bonding curves. It helps consumers discover high-quality assets/services. 
+* **Governance**: it governs the available assets and services in Ocean network. For example, it whitelists new service agreement templates through TCR & Voting. 
+* **Dispute Resolution**: resolves the disputes between consumer and provider through voting or TCR. 
+* **Block Reward**: Ocean network will periodically release block reward (i.e., new minted ERC20 Ocean tokens) to reward the providers of data commons. Note provider of data commons must be randomly selected to fulfill the service requests so that they have the equal probability to receive block rewards.
+* **Access Control**: it is the controller contract which grant access to consumers after depended conditions are fulfilled. The access token can be delivered in two approaches: secret store and OEP10. 
+* **Payment**: it handles all payment processing, including lock payment, release payment and refund payment. 
+
+
+### 1.2 Dependence
+
+The core modules have dependence on each other, which can be illustrated as following. Clearly, the `service agreement` module is the fundamental building block, where many other modules depends on it.  
+
+<img src="img/refactoring/dependence.jpg" width="1000" />
+
+
+## 2. Module Design
+
+In this section, each module is designed with details (some materials comes from [Research Guidance](https://docs.google.com/presentation/d/1Z6Acq2LD3eHPD1SoH_bxHH-XaOKLWemgFiKgWICekys/edit#slide=id.g44065f7377_2_1894)) and key smart contract functions are provided.
+
+### 2.1 Ocean Token 
+
+Ocean Token is the native and fungible tokens in Ocean network, which complies with the ERC20 Standard. 
+
+Its total supply of Ocean Token is 1.4 billion.
+
+45% of supply is reserved for block rewards, which will be minted according to schedule below:
+<img src="img/refactoring/schedule.jpg" width="500" />
+
+
+The key functions in Ocean token is similar to below:
+
+```Solidity
+string public constant name = 'OceanToken';                      // Set the token name for display
+string public constant symbol = 'OCN';                           // Set the token symbol for display
+
+// SUPPLY
+uint8 public constant decimals = 18;                             // Set the number of decimals for display
+uint256 public constant TOTAL_SUPPLY = 1400000000 * 10 ** 18;    // OceanToken total supply
+
+// transfer tokens 
+function transfer(address _to, uint256 _value) public returns (bool);

```

Comment:
This comment is slightly `implementation details` but it is recommended to use `msg.value` rather than using `unit256 _value`.

Reply:
We are using ERC20 token. `msg.value` is the amount of Ether, while `_value` is the amount of Ocean token. Here, transfer functions handles the transfer of Ocean tokens rather than Ether.